sys/net/gnrc/routing/rpl/mrhof.c

```diff
@@ -0,0 +1,288 @@
+{
+    if (stats == NULL) {
+        return MRHOF_MAX_PATH_COST;
+    }
+
+#if IS_USED(MODULE_GNRC_RPL_MRHOF_ETX)
+    return stats->etx;
+#elif IS_USED(MODULE_GNRC_RPL_MRHOF_LQI)
+    /* 255 = best, 0 = worst LQI; map best to ETX = 1, worst to ETX = 7 */
+    return 3 * (0xFF - stats->lqi) + NETSTATS_NB_ETX_DIVISOR;
+#else
+    return MRHOF_MAX_PATH_COST;
+#endif
+}
+
+/**
+ * Retrieve the full path cost of a parent
+ */
+static uint16_t _mrhof_get_path_cost(gnrc_rpl_parent_t *parent, netstats_nb_t *stats)
+{
+    return parent->rank + _link_metric(stats);
+}
+
+/**
+ * Check if a parent is within MAX_LINK_METRIC and MAX_PATH_COST
+ */
+static bool _mrhof_is_acceptable(gnrc_rpl_parent_t *parent, netstats_nb_t *stats)
+{
+    return _link_metric(stats) < MRHOF_MAX_LINK_METRIC &&
+           _mrhof_get_path_cost(parent, stats) < MRHOF_MAX_PATH_COST;
+}
+
+/**
+ * Check if a parent is fresh
+ */
+static bool _mrhof_isfresh(netif_t *netif, netstats_nb_t *stats)
+{
+    /* No statistics is not fresh */
+    if (stats == NULL) {
+        return false;
+    }
+    return netstats_nb_isfresh(netif, stats);
+}
+
+/* Compare acceptability of two parents */
+static int _mrhof_cmp_acceptable(gnrc_rpl_parent_t *p1, netstats_nb_t *p1_stats,
+                                 gnrc_rpl_parent_t *p2, netstats_nb_t *p2_stats)
+{
+    bool p1_acceptable = _mrhof_is_acceptable(p1, p1_stats);
+    bool p2_acceptable = _mrhof_is_acceptable(p2, p2_stats);
+
+    if (p1_acceptable == p2_acceptable) {
+        return 0;
+    }
+
+    /* One of the two parents is not acceptable */
+    if (p1_acceptable == false) {
+        /* p2 is acceptable and p1 not */
+        return 1;
+    }
+
+    if (p2_acceptable == false) {
+        /* p1 is acceptable and p2 not */
+        return -1;
+    }
+
+    return 0;
+}
+
+/* Compare freshness of two parents */
+static int _mrhof_cmp_fresh(netif_t *netif,
+                            netstats_nb_t *p1_stats,
+                            netstats_nb_t *p2_stats)
+{
+    bool p1_isfresh = _mrhof_isfresh(netif, p1_stats);
+    bool p2_isfresh = _mrhof_isfresh(netif, p2_stats);
+
+    if (p1_isfresh == p2_isfresh) {
+        return 0;
+    }
+
+    /* One of the two parents is not acceptable */
+    if (p1_isfresh == false) {
+        /* p2 is acceptable and p1 not */
+        return 1;
+    }
+
+    if (p2_isfresh == false) {
+        /* p1 is acceptable and p2 not */
+        return -1;
+    }
+
+    return 0;
+}
+
+static void reset(gnrc_rpl_dodag_t *dodag)
+{
+    /* Nothing to do in MRHOF */
+    (void) dodag;
+}
+
+/**
+ * Calculate rank additive based on the rank of the parent and the etx to the parent
+ * computed via MAX(pref_parent->rank, 1 + floor(MAX(parents->rank)/MinHopRankIncrease),
+ */
+static uint16_t calc_rank(gnrc_rpl_dodag_t *dodag, uint16_t base_rank)
+{
+    (void) base_rank;
+    DEBUG("MRHOF: Calculating rank\n");
+
+    /* TODO: consider a parent set of more than 1 parent */
```

Comment:
You are iterating through a list of parents below, so is this TODO maybe already done?

Reply:
@bergzand do you still remember why this was added?

Reply:
I have to admit that I really don't remember

Reply:
I'm fairly confident that it is outdated